src/librustc/traits/specialize/mod.rs

```diff
@@ -42,6 +42,7 @@ pub struct OverlapError {
     pub trait_desc: String,
     pub self_desc: Option<String>,
     pub intercrate_ambiguity_causes: Vec<IntercrateAmbiguityCause>,
+    pub used_to_be_allowed: bool,
```

Comment:
`used_to_be_allowed` is the name of a variable currently used in [librustc/traits/specialize/mod.rs](https://github.com/rust-lang/rust/blob/master/src/librustc/traits/specialize/mod.rs#L332).
This variable is setted based on the result of the insertion and is linked to the `IntercrateMode` used to check if two impls overalp. 
Since, with my changes, any `OverlapError`s is no more emitted at insertion time I keep track of this variable when creating the `OverlapError` struct and [i use it later](https://github.com/giannicic/rust/blob/intersection_impls/src/librustc/traits/specialize/mod.rs#L364) when checking if two impls overlap.